src/test/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchProcessResponseFilesImplTest.java

```diff
@@ -0,0 +1,256 @@
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.darts.arm.api.ArmDataManagementApi;
+import uk.gov.hmcts.darts.arm.config.ArmDataManagementConfiguration;
+import uk.gov.hmcts.darts.arm.model.blobs.ArmResponseBatchData;
+import uk.gov.hmcts.darts.arm.model.record.armresponse.ArmResponseCreateRecord;
+import uk.gov.hmcts.darts.arm.model.record.armresponse.ArmResponseInvalidLineRecord;
+import uk.gov.hmcts.darts.arm.model.record.armresponse.ArmResponseUploadFileRecord;
+import uk.gov.hmcts.darts.arm.service.ExternalObjectDirectoryService;
+import uk.gov.hmcts.darts.arm.util.files.BatchInputUploadFileFilenameProcessor;
+import uk.gov.hmcts.darts.arm.util.files.CreateRecordFilenameProcessor;
+import uk.gov.hmcts.darts.arm.util.files.InvalidLineFileFilenameProcessor;
+import uk.gov.hmcts.darts.arm.util.files.UploadFileFilenameProcessor;
+import uk.gov.hmcts.darts.authorisation.component.UserIdentity;
+import uk.gov.hmcts.darts.common.entity.ExternalObjectDirectoryEntity;
+import uk.gov.hmcts.darts.common.entity.ObjectStateRecordEntity;
+import uk.gov.hmcts.darts.common.helper.CurrentTimeHelper;
+import uk.gov.hmcts.darts.common.repository.ExternalObjectDirectoryRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectStateRecordRepository;
+import uk.gov.hmcts.darts.common.service.FileOperationService;
+import uk.gov.hmcts.darts.common.service.impl.EodHelperMocks;
+import uk.gov.hmcts.darts.common.util.EodHelper;
+import uk.gov.hmcts.darts.dets.config.DetsDataManagementConfiguration;
+import uk.gov.hmcts.darts.log.api.LogApi;
+
+import java.time.OffsetDateTime;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class DetsToArmBatchProcessResponseFilesImplTest {
+    @Mock
+    private ArmDataManagementApi armDataManagementApi;
+    @Mock
+    private ArmDataManagementConfiguration armDataManagementConfiguration;
+    @Mock
+    private ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    @Mock
+    private ExternalObjectDirectoryService externalObjectDirectoryService;
+    @Mock
+    private FileOperationService fileOperationService;
+    @Mock
+    private ObjectMapper objectMapper;
+    @Mock
+    private LogApi logApi;
+    @Mock
+    private DetsDataManagementConfiguration configuration;
+    @Mock
+    private ObjectStateRecordRepository osrRepository;
+    @Mock
+    private CurrentTimeHelper timeHelper;
+    @Mock
+    private UserIdentity userIdentity;
+
+    private DetsToArmBatchProcessResponseFilesImpl detsToArmBatchProcessResponseFilesImpl;
+
+    private static final EodHelperMocks EOD_HELPER_MOCKS = new EodHelperMocks();
```

Comment:
Should this be non-static as we would want the mocks to reset after each test?

Reply:
Its more for when we use verify as it will use the saved version for the counts, in this cause the static version which will never change.

so verify(EOD_HELPER_MOCKS.getMockedArmLocation(),times(1)).getId(), in this test and the same getId was called in a previous test the times called will be (2) but we only want it to be (1) as it has not reset the call count from the previous test as it is statically defined

Reply:
I don't need to verify the EOD_HELPER_MOCKS

Reply:
Its more if we do verify on other tests, if this runs first it will break the verify

Reply:
Again we should not be verifying the static test data.

Reply:
We may need to verify the returned result was called correctly to ensure we pass the correct values to other methods. For instance it had its getId called which we then pass to a repository, as these are all stored within your statically defined  EodHelperMocks the times it was called will be incorrect across tests

Reply:
So we can get this ticket merged in I have raised: https://tools.hmcts.net/jira/browse/DMP-4415 to look at this with a wider audience to gain a consensus on what the view is around static mocks